lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,255 @@
+    end
+
+    def subscribe_cluster(deployment_master, user)
+      perform_agent_commands(deployment_master, user, ["sudo mv ~/rhel_subscribe_inventory.yaml #{INVENTORIES_PATH}"])
+      run_playbook_command(deployment_master, user, "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
+                         "#{INVENTORIES_PATH}rhel_subscribe_inventory.yaml 1> #{ANSIBLE_LOG} 2> #{ANSIBLE_ERROR_LOG} < /dev/null")
+    end
+
+    def provision_tagged_tasks
+      tasks = []
+      MiqRequest.all.each do |prov|
+        next unless prov.get_tags[:miq_openshift_deploy] &&
+                    (prov.get_tags[:miq_openshift_deploy].include?(get_custom_tag("master")) ||
+                      prov.get_tags[:miq_openshift_deploy].include?(get_custom_tag("node")))
+        tasks << prov
+      end
+      MiqAeServiceModelBase.wrap_results(tasks)
+    end
+
+    def get_custom_tag(type)
+      type + "_#{automation_task.id}"
+    end
+
+    def nodes_subscription_needed?
+      container_deployment_nodes.count > 1
+    end
+
+    def ssh_user
+      ssh_auth.userid
+    end
+
+    def add_deployment_provider(options)
+      add_deployment_provider(options)
+    end
+
+    def regenerate_ansible_inventory
+      generate_ansible_yaml
+    end
+
+    def regenerate_ansible_subscription_inventory
+      generate_ansible_inventory_for_subscription
+    end
+
+    def add_automation_task(task)
+      ar_method do
+        @object.automation_task = AutomationTask.find_by_id(task.id)
+        @object.save!
+      end
+    end
+
+    def customize(options = {})
+      ar_method do
+        if @object.customizations.empty?
+          @object.customizations = {:agent => {}}
+        end
+        @object.customizations[:agent].merge!(options)
+        @object.save!
+      end
+    end
+
+    def perform_scp(ip, username, local_path, remote_path)
+      Net::SCP.upload!(ip, username, local_path, remote_path, :ssh => {:key_data => ssh_auth.auth_key})
+    end
+
+    def perform_agent_commands(ip, username, commands = [])
+      @ssh ||= LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      @ssh.perform_commands(commands)
+    end
+
+    def provision_vms_status
+      fail = false
+      number_of_finished_vms = 0
+      tasks = provision_tagged_tasks
+      tasks.each do |prov|
+        if prov.request_state.include?('finished')
+          number_of_finished_vms += 1 if prov.status.include?('Ok')
+          fail = true if prov.status.include?('Error')
+        end
+      end
+      [number_of_finished_vms, fail]
+    end
+
+    def find_vm_by_tag(tag, no_wrapper = true)
+      tag = "/managed/miq_openshift_deploy/#{tag}_#{automation_task.id}"
+      return Vm.find_tagged_with(:any => tag, :ns => "*") if no_wrapper
+      wrap_results(Vm.find_tagged_with(:any => tag, :ns => "*"))
```

Comment:
Use the stored request IDs from the automation task to load the VMs.
```ruby
def find_vm_by_tag(type)
  requests = MiqRequest.where(:id => automation_task.miq_request.options["#{tag}_request_ids"])
  requests.collect(&:vms).flatten
end
```

With these changes are the tags still required?